scripts/src/javatests/org/oppia/android/scripts/ci/ComputeChangedFilesTest.kt

```diff
@@ -0,0 +1,826 @@
+    createFiles("NewUntrackedFile", subPackage = "data")
+
+    val reportedFiles = runScript()
+
+    // Only the first file should be reported since the file itself was changed & staged.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("data/NewUntrackedFile.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_deletedFile_committed_returnsNoFiles() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", subPackage = "app")
+    switchToFeatureBranch()
+    removeAndCommitFile("First", subPackage = "app")
+
+    val reportedFiles = runScript()
+
+    // Removing the file should result in no files being returned (since the file is gone).
+    assertThat(reportedFiles).isEmpty()
+  }
+
+  @Test
+  fun testUtility_featureBranch_movedFile_staged_returnsNewFile() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", subPackage = "app")
+    switchToFeatureBranch()
+    moveFile(
+      oldFileName = "First",
+      oldSubPackage = "app",
+      newFileName = "RenamedFile",
+      newSubPackage = "domain"
+    )
+
+    val reportedFiles = runScript()
+
+    // The file should show up under its new name since moving it is the same as changing it.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("domain/RenamedFile.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_multipleFilesChanged_committed_returnsChangedFiles() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+    changeAndCommitFile("First", subPackage = "app")
+    changeAndCommitFile("Third", subPackage = "app")
+
+    val reportedFiles = runScript()
+
+    // Changing multiple files independently should be reflected in the script's results.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList)
+      .containsExactly("app/First.kt", "app/Third.kt")
+  }
+
+  @Test
+  fun testUtility_developBranch_instrumentationModuleChanged_instrumentationFilesAreIgnored() {
+    initializeEmptyGitRepository()
+    createFiles(
+      "InstrumentationFile",
+      subPackage = "instrumentation/src/javatests/org/oppia/android/instrumentation/player"
+    )
+    createFiles(
+      "Robolectric",
+      subPackage = "instrumentation/src/javatests/org/oppia/android/instrumentation/app"
+    )
+    createFiles(
+      "Third",
+      subPackage = "instrumentation"
+    )
+
+    val reportedFiles = runScript()
+
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).doesNotContain(
+      "instrumentation/src/javatests/org/oppia/android/" +
+        "instrumentation/player/InstrumentationFile.kt"
+    )
+    assertThat(reportedFiles.first().changedFilesList).contains(
+      "instrumentation/src/javatests/org/oppia/android/instrumentation/app/Robolectric.kt"
```

Comment:
Isn't this supposed to be ignored (per the expected behavior & the name of the test)?

Reply:
The filter only filters out files in the path - "instrumentation/src/javatests/org/oppia/android/instrumentation/player", so since instrumentation/src/javatests/org/oppia/android/instrumentation/app/Robolectric.kt follows a different path, it gets included.

I realize the test name is misleading and maybe I can split them as 2 test cases specifically for their own conditions.

Reply:
That helps--thanks!